Formula/codelite.rb

```diff
@@ -0,0 +1,35 @@
+class Codelite < Formula
+  desc "cross platform C/C++/PHP and Node.js IDE written in C++"
+  homepage "https://codelite.org"
+  url "https://github.com/eranif/codelite/archive/11.0.tar.gz"
+  sha256 "9eb23ea635fc746318a832efd752b08027c11efcd9af7f494f31dbaacdd651a1"
+
+  depends_on "ccache" => :build
+  depends_on "cmake" => :build
+  depends_on "openssl"
+  depends_on "wxmac"
+
+  def install
+    # https://github.com/eranif/codelite/issues/1284
+    # https://github.com/eranif/codelite/pull/1797
+    inreplace "CMakeLists.txt", "set(WX_COMPONENTS \"std\")", "set(WX_COMPONENTS \"std aui propgrid ribbon\")"
+    inreplace "codelite_terminal/CMakeLists.txt", "wxWidgets REQUIRED", "wxWidgets COMPONENTS std stc REQUIRED"
+    mkdir "build-release" do
+      system "cmake", "..", "-DCL_PREFIX=#{prefix}", *std_cmake_args
+      system "make", "install"
+      mkdir prefix/"libexec"
+      prefix.install "codelite.app"
+    end
+  end
+
+  def caveats; <<~EOS
+      Known Issue:
+      If you built `wxmac` with "--with-stl" option, you cannot build this Formula.
```

Comment:
Given most people don't reach for `brew info` to resolve curiosity about new formulae, it's likely by the time most people sees this either they'll have poured the bottle & it'll be irrelevant, or their build will have blown up and they'll go looking for why. I'm not sure this provides value as-is, personally.